vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/AbstractNumberField.java

```diff
@@ -99,6 +101,35 @@ public AbstractNumberField(SerializableFunction<String, T> parser,
         });
     }
 
+    @Override
+    public void setValueChangeMode(ValueChangeMode valueChangeMode) {
+        if (inputListenerRegistration != null) {
+            inputListenerRegistration.remove();
+            inputListenerRegistration = null;
+        }
+
+        if (valueChangeMode.equals(ValueChangeMode.EAGER)
+                || valueChangeMode.equals(ValueChangeMode.LAZY)
+                || valueChangeMode.equals(ValueChangeMode.TIMEOUT)) {
```

Comment:
nit: could use Yoda conditions here to avoid a possible NPE (`ValueChangeMode.EAGER.equals(valueChangeMode)`)

Reply:
Thanks, updated.